src/tbb/task_dispatcher.h

```diff
@@ -338,6 +338,7 @@ d1::task* task_dispatcher::local_wait_for_all(d1::task* t, Waiter& waiter ) {
 
                     ITT_CALLEE_ENTER(ITTPossible, t, itt_caller);
 
+                    m_innermost_running_task = t;
```

Comment:
Same double setting of m_innermost_running_task issue.

Reply:
Rebased to part1 after merging the transferring support. Resolved.